boskos/ranch/ranch_test.go

```diff
@@ -455,6 +455,9 @@ func TestRelease(t *testing.T) {
 			if !AreErrorsEqual(releaseErr, tc.expectErr) {
 				t.Fatalf("Got error %v, expected error %v", releaseErr, tc.expectErr)
 			}
+			if tc.expectedRes != nil && tc.expectedRes.Status.UserData == nil {
```

Comment:
couldn't we fix this by making `crds.NewResource` always set UserData to an empty struct?

Reply:
Yeah, but then we need to set it on other tests so they keep passing :) I did it anyways, mostly because the code seems to assume that its not nil and calls methods on it that will just panic if its nil. I did it in a separate commit, tell me what you think.